### ISSUE TITLE
Add support for importantForAccessibility="no-hide-descendants" or equivalent

### ISSUE THREAD

**Issue by aschultz** [microsoft/react-native-windows#5113]
# Proposal: Add property to recursively hide elements from screen readers

## Summary
Android (and C# RNW) supports `importantForAccessibility="no-hide-descendants"` to recursively hide an element and its children from screen readers. iOS has `accessibilityElementsHidden=true`, and web has `aria-hidden=true`, to accomplish the same thing.

React Native Windows (C++) should support a similar recursive property.

## Motivation
When creating layered visual content (a slide out tray, a modal-like popup), often the background content should not be interactive or accessible while the foreground element exists.

If the foreground is completely opaque, this might be accomplished by not rendering the background at all. However, if any part of the background should remain visible (for example, through an acrylic panel in the foreground), another mechanism is needed to disable the interaction and screen reader access.

The "accessible=false" property allows this on an element-by-element level (in the case of RNW, it sets AccessibilityView::Raw), but it doesn't recursively apply this to children, which makes it a poor fit for hiding larger chunks of content. 

## Open Questions
- Should this use one of the existing properties (importantForAccessibility or accessibilityElementsHidden) or a new windows-specific property?

**Comment by chrisglein** [comment]
Seems like lack of unification across RN platforms here. We don't pile on with a different solution, but also it seems like we have to pick one to support. There's precedence given vLegacy used the Android API (and ignored the iOS one), so we could match what was done there. But we're going to want to upstream some kind of unity on this.

We should look at what the C# legacy version did to start with, but continue the conversation of how to get unification in RN core on this.

**Comment by asklar** [comment]
Support for `no-hide-descendants` was added in #1807

**Comment by chrisglein** [comment]
@AgneLukoseviciute can you look into the complexity of this (given the available C# implementation) and give a recommendation on what it'd cost to fill this gap?

**Comment by aschultz** [comment]
I'm not familiar with the old C# implementation, but we have an implementation in Garrison that may be useful for reference, at least for "no-hide-descendants". It overrides AutomationPeer::GetChildrenCore() and calls InvalidatePeer() to hide a subtree from Narrator.

**Comment by chiaramooney** [comment]
@aschultz Would it be possible to send me a link (offline or here) to the source for the Garrison implementation of "no-hide-descendants", so we can check that out as a solution possibility?

**Comment by AgneLukoseviciute** [comment]
Reopening since we still have some work to do here, working on a native implementation.

**Comment by chiaramooney** [comment]
Will not create native implementation on Paper. Will flag to implement on Fabric.
